packages/cli/src/api/formats/po-gettext.ts

```diff
@@ -0,0 +1,256 @@
+import { format as formatDate } from "date-fns"
+import fs from "fs"
+import ICUParser from "messageformat-parser"
+import pluralsCldr from "plurals-cldr"
+import PO from "pofile"
+import querystring from "querystring"
+import * as R from "ramda"
+
+import { CatalogType, MessageType } from "../types"
+import { joinOrigin, splitOrigin, writeFileIfChanged } from "../utils"
+
+
+// Workaround because pofile doesn't support es6 modules, see https://github.com/rubenv/pofile/pull/38#issuecomment-623119284
+type POItemType = InstanceType<typeof PO.Item>
+
+function getCreateHeaders(language = "no") {
+  return {
+    "POT-Creation-Date": formatDate(new Date(), "yyyy-MM-dd HH:mmxxxx"),
+    "Mime-Version": "1.0",
+    "Content-Type": "text/plain; charset=utf-8",
+    "Content-Transfer-Encoding": "8bit",
+    "X-Generator": "@lingui/cli",
+    Language: language,
+  }
+}
+
+// Attempts to turn a single tokenized ICU plural case back into a string.
+function stringifyICUCase(icuCase) {
+  return icuCase.tokens
+    .map((token) => {
+      if (typeof token === "string") {
+        return token
+      } else if (token.type === "octothorpe") {
+        return "#"
+      } else if (token.type === "argument") {
+        return "{" + token.arg + "}"
+      } else {
+        console.warn(
+          `Unexpected token "${token}" while stringifying plural case "${icuCase}". Token will be ignored.`
+        )
+        return ""
+      }
+    })
+    .join("")
+}
+
+const serialize = (items: CatalogType, options) => R.compose(
+  R.values,
+  R.mapObjIndexed((message: MessageType, key) => {
+    const item = new PO.Item()
+    item.msgid = key
+
+    // Depending on whether custom ids are used by the developer, the (potential plural) "original", untranslated ICU
+    // message can be found in `message.message` or in the item's `key` itself.
+    const icuMessage = message.message || key
+
+    const _simplifiedMessage = icuMessage.replace(/\n|\r\n/g, " ")
+
+    // Quick check to see if original message is a plural localization.
+    if (/^{.*, plural, .*}$/.test(_simplifiedMessage)) {
+      try {
+        const [messageAst] = ICUParser.parse(icuMessage)
+        
+        // Check if any of the plural cases contain plurals themselves.
+        if (messageAst.cases.some((icuCase) => icuCase.tokens.some((token) => token.type === "plural"))) {
+          console.warn(`Nested plurals cannot be expressed with gettext plurals. ` +
+            `Message with key "%s" will not be saved correctly.`, key)
+        }
+
+        // Store placeholder that is pluralized upon to allow restoring ICU format later.
+        item.msgctxt = querystring.encode({ pluralize_on: messageAst.arg })
```

Comment:
With the new additions of today, I need to store one or two items in the message context and wanted to use something that returns an object after parsing.
I switched to using `URLSearchParams` (which has been available since Node 10 which is the oldest currently supported version), only to find out that `querystring` is also a built-in library (I thought I could eliminate a 3rd party dependency).

Well, at least it's a WhatWG standard...